tenets/codelingo/kubernetes/misused-nil-interface/codelingo.yaml

```diff
@@ -13,69 +19,30 @@ tenets:
         body: |
           Nil interface is not the same as having a nil pointer in the interface.
       codelingo/review:
-        comment: The variable being tested for nil, `{{varname}}` is a pointer to a pointer to nil.
+        comment: The value of the returned variable is not nil! The interface `{{interfaceName}}` serves as a pointer, therefore the returned variable is a pointer to a nil pointer.
     query: |
       import codelingo/ast/go
       
-      go.decls:
-        go.gen_decl:
-          go.type_spec:
-            go.ident:
-              type as interfacetype
-            go.interface_type:
-              go.field_list:
-                go.field:
-                  go.names:
-                    go.ident:
-                      name as purevirtualfuncname
-                      type == "func()"
-        go.gen_decl:
-          go.type_spec:
-            go.ident:
-              type as typename
-            go.struct_type
-        go.func_decl:
-          go.field_list:
-            go.field:
-              go.star_expr:
-                go.ident:
-                  type == typename
+      # Note: This is only going to find instances where the interface is defined inside the file.
+      # so won't pick up https://golang.org/doc/faq#nil_error.
+      go.decls(depth = any):
+        go.type_spec(depth = any):
           go.ident:
-            name == purevirtualfuncname
-          go.func_type
+            name as interfaceName
+            type as interfaceType
+          go.interface_type
         go.func_decl:
-          go.ident:
-            name as factoryfuncname
           go.func_type:
-            go.field_list
             go.field_list:
```

Comment:
Change to

```
          go.func_type:
            go.field_list
            go.field_list:
```

Otherwise it will also match any function with a parameter called interfaceType, not just return parameters